src/Mappers/BaseMapper.php

```diff
@@ -62,17 +64,158 @@ protected static function getValueFromTag(\DOMDocument $document, string $tag):
         return $element->textContent;
     }
 
-    protected static function getField(\DOMElement $element, string $fieldTagName): ?string
+    protected static function getField(\DOMElement $element, string $fieldTagName, $object = null): ?string
     {
         $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
         if (!isset($fieldElement)) {
             return null;
         }
 
+        if (isset($object)) {
+            self::checkForMessage($object, $fieldElement);
+        }
+
         if ($fieldElement->textContent === "") {
             return null;
         }
 
         return $fieldElement->textContent;
     }
+    
+    protected static function checkForMessage($object, \DOMElement $element): void
+    {
+        if ($element->hasAttribute('msg')) {
+            $message = new Message();
+            $message->setType($element->getAttribute('msgtype'));
+            $message->setMessage($element->getAttribute('msg'));
+            $message->setField($element->nodeName);
+
+            $object->addMessage($message);
+        }
+    }
+
+    protected static function getAttribute(\DOMElement $element, string $fieldTagName, string $attributeName): ?string
+    {
+        $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
+        if (!isset($fieldElement)) {
+            return null;
+        }
+
+        if ($fieldElement->getAttribute($attributeName) === "") {
+            return null;
+        }
+
+        return $fieldElement->getAttribute($attributeName);
+    }
+
+    protected static function parseBooleanAttribute(?string $value): ?bool
+    {
+        return filter_var($value, FILTER_VALIDATE_BOOLEAN);
+    }
+
+    protected static function parseDateAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDate($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseDateTimeAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDateTime($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseEnumAttribute(string $enumName, ?string $value)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        $enum = "\\PhpTwinfield\\Enums\\" . $enumName;
+
+        try {
+            $classReflex = new \ReflectionClass($enum);
+            $classConstants = $classReflex->getConstants();
+
+            foreach ($classConstants as $classConstant) {
+                if ($value == $classConstant) {
```

Comment:
Should use strict equality, Enum, supports 0, false and null as valid values.

Reply:
Alternatively, you can just try to construct them and if it fails, return null (or fail).

Reply:
Changed to:

`try {
            return new $enumClass($value);
        } catch (\Exception $e) {
             return null;
        }`